cmd/cosign/cli/generate/generate_key_pair.go

```diff
@@ -57,8 +60,20 @@ func GenerateKeyPairCmd(ctx context.Context, kmsVal string, args []string) error
 		fmt.Fprintln(os.Stderr, "Public key written to cosign.pub")
 		return nil
 	}
+
 	if len(args) > 0 {
-		return kubernetes.KeyPairSecret(ctx, args[0], GetPass)
+		// github://repository/project
+		split := strings.Split(args[0], "://")
+		provider, targetRef := split[0], split[1]
+
+		switch provider {
+		case "k8s":
+			return kubernetes.KeyPairSecret(ctx, targetRef, GetPass)
+		case gitlab.GitLabReference,github.GitHubReference:
+			return git.GetGit(provider).PutSecret(ctx, targetRef, GetPass)
+		}
+
+		return fmt.Errorf("undefined provider")
```

Comment:
Better error out would be: `fmt.Errorf("undefined provider: %s", provider)`

Reply:
thank you, fixed